core/database/src/main/java/de/mm20/launcher2/database/migrations/Migration_21_22.kt

```diff
@@ -0,0 +1,13 @@
+package de.mm20.launcher2.database.migrations
+
+import androidx.room.migration.Migration
+import androidx.sqlite.db.SupportSQLiteDatabase
+
+class Migration_21_22: Migration(21, 22) {
+    override fun migrate(database: SupportSQLiteDatabase) {
+        database.execSQL("""
+            ALTER TABLE `Searchable`
+            ADD `weight` DOUBLE NOT NULL DEFAULT 0.0
+            """)
+    }
+}
```

Comment:
Is this migration fine? I had an issue where the launcher crashed with an exception related to migration:

<details>
<summary>Stacktrace</summary>

```
java.lang.IllegalStateException: Migration didn't properly handle: Searchable(de.mm20.launcher2.database.entities.SavedSearchableEntity).
                                                                                                     Expected:
                                                                                                    TableInfo{name='Searchable', columns={launchCount=Column{name='launchCount', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, pinned=Column{name='pinned', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, hidden=Column{name='hidden', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, weight=Column{name='weight', type='REAL', affinity='4', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, type=Column{name='type', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, key=Column{name='key', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=1, defaultValue='undefined'}, searchable=Column{name='searchable', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}}, foreignKeys=[], indices=[]}
                                                                                                     Found:
                                                                                                    TableInfo{name='Searchable', columns={key=Column{name='key', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=1, defaultValue='undefined'}, type=Column{name='type', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, searchable=Column{name='searchable', type='TEXT', affinity='2', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, launchCount=Column{name='launchCount', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, pinned=Column{name='pinned', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, hidden=Column{name='hidden', type='INTEGER', affinity='3', notNull=true, primaryKeyPosition=0, defaultValue='undefined'}, weight=Column{name='weight', type='DOUBLE', affinity='4', notNull=false, primaryKeyPosition=0, defaultValue='0.0'}}, foreignKeys=[], indices=[]}
                                                                                                    	at androidx.room.RoomOpenHelper.onUpgrade(RoomOpenHelper.kt:94)
                                                                                                    	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.onUpgrade(FrameworkSQLiteOpenHelper.kt:252)
                                                                                                    	at android.database.sqlite.SQLiteOpenHelper.getDatabaseLocked(SQLiteOpenHelper.java:416)
                                                                                                    	at android.database.sqlite.SQLiteOpenHelper.getWritableDatabase(SQLiteOpenHelper.java:316)
                                                                                                    	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getWritableOrReadableDatabase(FrameworkSQLiteOpenHelper.kt:231)
                                                                                                    	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.innerGetDatabase(FrameworkSQLiteOpenHelper.kt:189)
                                                                                                    	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper$OpenHelper.getSupportDatabase(FrameworkSQLiteOpenHelper.kt:151)
                                                                                                    	at androidx.sqlite.db.framework.FrameworkSQLiteOpenHelper.getWritableDatabase(FrameworkSQLiteOpenHelper.kt:104)
                                                                                                    	at androidx.room.RoomDatabase.inTransaction(RoomDatabase.kt:629)
                                                                                                    	at androidx.room.RoomDatabase.assertNotSuspendingTransaction(RoomDatabase.kt:448)
                                                                                                    	at androidx.room.RoomDatabase.query(RoomDatabase.kt:477)
                                                                                                    	at androidx.room.util.DBUtil.query(DBUtil.kt:75)
                                                                                                    	at de.mm20.launcher2.database.SearchDao_Impl$23.call(SearchDao_Impl.java:972)
                                                                                                    	at de.mm20.launcher2.database.SearchDao_Impl$23.call(SearchDao_Impl.java:969)
                                                                                                    	at androidx.room.CoroutinesRoom$Companion$createFlow$1$1$1.invokeSuspend(CoroutinesRoom.kt:128)
                                                                                                    	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
                                                                                                    	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1137)
                                                                                                    	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:637)
                                                                                                    	at java.lang.Thread.run(Thread.java:1012)
                                                                                                    	Suppressed: kotlinx.coroutines.DiagnosticCoroutineContextException: [StandaloneCoroutine{Cancelling}@7542527, Dispatchers.Main.immediate]
```
</details>

Uninstalling Kvaesitso and redeploying fixed this, but I think this error will reoccur.

Reply:
I think you need to annotate the entity's field with [@ColumnInfo(defaultValue = 0.0)](https://developer.android.com/reference/androidx/room/ColumnInfo.html#defaultValue()) in order to make the schemas match.

Reply:
I've checked by deploying current main and then installing this branch's version, and this seems to be fine now. Maybe you also check to make sure?

Reply:
Seems to be working